src/main/java/com/example/restexample/controller/PatientController.java

```diff
@@ -0,0 +1,64 @@
+package com.example.restexample.controller;
+
+
+import com.example.restexample.dto.PatientDto;
+import com.example.restexample.exception.DoctorNotFoundException;
+import com.example.restexample.exception.PatientNotFoundException;
+import com.example.restexample.service.PatientService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/patients")
+public class PatientController {
+    private final PatientService patientService;
+
+
+    @Autowired
+    public PatientController(PatientService patientService) {
+        this.patientService = patientService;
+
+    }
+
+
+    @GetMapping
+    public List<PatientDto> getAllPatients() {
+        return patientService.findAll();
+    }
+
+    @GetMapping("/doctors/{id}")
+    public List<PatientDto> getAllPatientsByDoctorId(@PathVariable(name = "id") Integer id) {
```

Comment:
/patients/doctors/{id} ? Hard to understand the meaning of the link and resources